zlib/grabber.go

```diff
@@ -213,6 +215,15 @@ func containsPort(host string) bool {
 	return strings.LastIndex(host, ":") > strings.LastIndex(host, "]")
 }
 
+func redirectsToLocalhost(host string) bool {
+	if len(host) >= 3 {
+		if host[:3] == "127" || host[:3] == "::1" || host == "localhost" {
```

Comment:
You should actually convert to an IP and check against subnet masks.

Also, does this work if the host if `example.com` but it resolves to `127.0.0.1`?